### DISCUSSION TITLE
Firebase error on Cordova app

### DISCUSSION THREAD

**Discussion by i-mage** [quasarframework/quasar#12167] [CLI - Capacitor / Cordova modes]
I have an issue with Firebase OAuth. While testing the app in my broswer, the method firebase.auth().signInWithPopup(provider) worked as expected. However, when I build the app in Cordova, I receive the following error when I try to use Firebase:

`error Error: This operation is not supported in the environment this application is running on. "location.protocol" must be http, https or chrome-extension and web storage must be enabled.`

Any hint how to solve this issue would be greatly appreciated.

**Comment by i-mage** [comment]
So I have moved forward and replaced singInWithPopup with signInWithRedirect according to https://firebase.google.com/docs/auth/web/cordova

Again, this works smoothly in the browser. In the app, I get the Google login screen and after I log in and being redirected back to app I get the following error - Safari cannot open the page because the address is invalid.

**Comment by MilosPaunovic** [comment]
Take a look at this: [https://github.com/quasarframework/quasar/discussions/11514](https://github.com/quasarframework/quasar/discussions/11514).

**Reply by i-mage** [reply]
So I suppose I need to use one of the following plugins to get this work:

https://github.com/EddyVerbruggen/cordova-plugin-googleplus
https://github.com/cordova-plugin-facebook-connect/cordova-plugin-facebook-connect
https://devdactic.com/capacitor-google-sign-in/
